src/storage-client/src/controller.rs

```diff
@@ -1916,8 +1916,15 @@ where
                     export.write_frontier = new_upper.clone();
                 }
 
-                let mut new_read_capability =
-                    export.read_policy.frontier(export.write_frontier.borrow());
+                // Ignore read policy for sinks whose write frontiers are closed, which identifies
+                // the sink is being dropped; we need to advance the read frontier to the empty
+                // chain to signal to the dataflow machinery that they should deprovision this
+                // object.
+                let mut new_read_capability = if export.write_frontier.is_empty() {
+                    export.write_frontier.clone()
```

Comment:
This piece is what makes sure that we emit an `AllowCompaction([])` for the sink id, right?

Reply:
yes